annotated.go

```diff
@@ -264,6 +264,7 @@ type annotated struct {
 	ParamTags  []string
 	ResultTags []string
 	As         [][]reflect.Type
+	FuncInfo   dig.ProvideOption
```

Comment:
Can this be *
```suggestion
	FuncPtr   uintptr
```

Annotated should keep track of the minimum info needed to let Dig know where this function is does not need to keep track of the whole function.